### ISSUE TITLE
LINE算法模型训练相关问题咨询

### ISSUE THREAD

**Issue by ivankxt** [alibaba/euler#98]
样本数据量：大概9千万个节点，node_id（0~90000000），机器内存251G，64核，1ps + 5worker

bash dist_tf_euler.sh --data_dir hdfs://path/data   --euler_zk_addr IP:2181 --euler_zk_path /euler   --max_id 90000000 --model line --mode train --model_dir ./model  --learning_rate 0.01 --dim 512

1、运行line算法模型时，设置训练参数： --max_id 90000000，--dim 512，抛出如下错误：
2019-04-28 09:53:19.518423: W tensorflow/core/framework/allocator.cc:124] Allocation of 184231307264 exceeds 10% of system memory.
2019-04-28 09:53:19.518480: W tensorflow/core/framework/op_kernel.cc:1401] OP_REQUIRES failed at random_op.cc:202 : Resource exhausted: OOM when allocating tensor with shape[89956693,512] and type float on /job:ps/replica:0/task:0/device:CPU:0 by allocator cpu
请问这个问题也没有办法解决？

2、将dim设置为--dim 12时，某个worker打印日志如下：
INFO:tensorflow:loss = 2129.7446, mrr = 0.40621746, step = 0
INFO:tensorflow:loss = 14377.218, mrr = 0.44117844, step = 84 (232.371 sec)
INFO:tensorflow:loss = 127506900.0, mrr = 0.4401042, step = 187 (256.154 sec)
......
通过观察发现，ps进程占用的cpu达到4000%左右，而对应的worker占用cpu不到1%，即所有压力落在的ps上，请问如何提升worker的计算能力及降低ps的压力？
另外，我们将ps数增大为3个时，即3ps + 5worker，主ps占用的cpu依旧很大（4000%左右），而其他两个ps几乎不占用cpu，并且worker占用cpu也不到1%，请问这如何解释？

3、在问题2的条件下（2ps + 5worker），运行时会出现如下问题：
INFO:tensorflow:loss = 2128.9492, mrr = 0.4132162, step = 0
INFO:tensorflow:loss = 144028.48, mrr = 0.41656902, step = 88 (227.890 sec)
INFO:tensorflow:loss = 21924712.0, mrr = 0.43326825, step = 195 (236.491 sec)
INFO:tensorflow:An error was raised. This may be due to a preemption in a connected worker or parameter server. The current session will be closed and a new session will be created. This error may also occur due to a gRPC failure caused by high memory or network bandwidth usage in the parameter servers. If this error occurs repeatedly, try increasing the number of parameter servers assigned to the job. Error: Socket closed
请问这是什么原因引起的？

谢谢解答~

**Comment by qyhfbqz** [comment]
1.9000w*512维的矩阵大概要300多G内存，解决办法就是减小矩阵的大小
2.分布式的话参数需要需要分片

**Comment by ivankxt** [comment]
> 1.9000w*512维的矩阵大概要300多G内存，解决办法就是减小矩阵的大小
> 2.分布式的话参数需要需要分片

hi, @qyhfbqz 请问，参数分片具体怎么做？数据分片我们知道，但不太清楚参数分片是怎么玩？

**Comment by ShangJP** [comment]
@ivankxt 您好，将dim设置为--dim 12时，某个worker打印日志如下：
INFO:tensorflow:loss = 2129.7446, mrr = 0.40621746, step = 0
INFO:tensorflow:loss = 14377.218, mrr = 0.44117844, step = 84 (232.371 sec)
INFO:tensorflow:loss = 127506900.0, mrr = 0.4401042, step = 187 (256.154 sec)，我看到您在训练的过程中loss 也是异常的增大，请问您知道这是什么原因么

**Comment by alinamimi** [comment]
#176
